cmd/domains_cmd.go

```diff
@@ -0,0 +1,321 @@
+	svc, formatter := WireUpDomain(c)
+
+	checkRequiredFlags(c, []string{"id"}, formatter)
+	domain, err := svc.DeleteDomain(c.String("id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't delete dns domain", err)
+	}
+
+	_, labelNamesByID := LabelLoadsMapping(c)
+	domain.FillInLabelNames(labelNamesByID)
+	if err = formatter.PrintItem(*domain); err != nil {
+		formatter.PrintFatal("Couldn't print/format result", err)
+	}
+	return nil
+}
+
+// DomainRetry subcommand function
+func DomainRetry(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	svc, formatter := WireUpDomain(c)
+
+	checkRequiredFlags(c, []string{"id"}, formatter)
+
+	domain, err := svc.RetryDomain(c.String("id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't retry dns domain", err)
+	}
+
+	_, labelNamesByID := LabelLoadsMapping(c)
+	domain.FillInLabelNames(labelNamesByID)
+	if err = formatter.PrintItem(*domain); err != nil {
+		formatter.PrintFatal("Couldn't print/format result", err)
+	}
+	return nil
+}
+
+// DomainListRecords subcommand function
+func DomainListRecords(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	svc, formatter := WireUpDomain(c)
+
+	checkRequiredFlags(c, []string{"domain-id"}, formatter)
+	records, err := svc.ListRecords(c.String("domain-id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't receive dns records data", err)
+	}
+	if err = formatter.PrintList(records); err != nil {
+		formatter.PrintFatal("Couldn't print/format result", err)
+	}
+	return nil
+}
+
+// DomainShowRecord subcommand function
+func DomainShowRecord(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	svc, formatter := WireUpDomain(c)
+
+	checkRequiredFlags(c, []string{"id"}, formatter)
+	record, err := svc.GetRecord(c.String("id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't receive dns record data", err)
+	}
+	if err = formatter.PrintItem(*record); err != nil {
+		formatter.PrintFatal("Couldn't print/format result", err)
+	}
+	return nil
+}
+
+// DomainCreateRecord subcommand function
+func DomainCreateRecord(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	svc, formatter := WireUpDomain(c)
+
+	checkRequiredFlags(c, []string{"domain-id", "name", "type"}, formatter)
+	recordType := c.String("type")
+	recordIn := map[string]interface{}{
+		"name": c.String("name"),
+		"type": recordType,
+	}
+
+	if c.IsSet("content") {
+		recordIn["content"] = c.String("content")
+	}
+	if c.IsSet("ttl") {
+		recordIn["ttl"] = c.Int("ttl")
+	}
+
+	// If provided, only include in adequate context
+	switch recordType {
+	case "a":
+		// one and only one of the fields must be provided.
+		if c.IsSet("server-id") && c.IsSet("floating-ip-id") {
+			return fmt.Errorf("invalid parameters detected. Please provide only one: 'server-id' or 'floating-ip-id'")
+		}
+		if c.IsSet("server-id") {
+			recordIn["instance_id"] = c.String("server-id")
+		}
+		if c.IsSet("floating-ip-id") {
+			recordIn["floating_ip_id"] = c.String("floating-ip-id")
+		}
+	case "cname":
+		if c.IsSet("load-balancer-id") {
```

Comment:
I'd also return an error here if content is set and load balancer id is set, with an error like: ' invalid parameters detected. Please provide only one: content or load-balancer-id'

Reply:
Done. 
Additionally, it seems that API request accepts both (no error) but does not apply given content. It seems calculated based on the load balancer